lib/src/config/config.rs

```diff
@@ -560,6 +711,67 @@ impl ClientConfigBuilder {
                 self.reverse_proxy = Some(Some(reverse_proxy.clone().into()));
             });
 
+        // Configure RPC server
+        #[cfg(feature="rpc-server")] {
+            if let Some(rpc_config) = &config_file.rpc_server {
+                let bind_to = rpc_config.bind.as_ref()
+                    .and_then(|addr| addr.into_ip_address());
+
+                let allow_ips = if rpc_config.allowip.is_empty() {
+                    None
+                }
+                else {
+                    let result = rpc_config.allowip.iter().map(|s| {
+                        s.parse::<IpAddr>().map_err({
+                            |e| Error::config_error(format!("Invalid IP: {}", e))
+                        })
+                    }).collect::<Result<Vec<IpAddr>, Error>>();
+                    Some(result?)
+                };
+
+                let credentials = match (&rpc_config.username, &rpc_config.password) {
+                    (Some(u), Some(p)) => {
+                        Some(Credentials::new(u.clone(), p.clone()))
+                    },
+                    (None, None) => None,
+                    _ => return Err(Error::config_error("Either both username and password have to be set or none."))
+                };
+
+                self.rpc_server = Some(Some(RpcServerConfig {
+                    bind_to,
+                    port: rpc_config.port.unwrap_or(consts::RPC_DEFAULT_PORT),
+                    corsdomain: Some(rpc_config.corsdomain.clone()),
+                    allow_ips,
+                    allowed_methods: Some(rpc_config.methods.clone()),
+                    credentials,
+                }));
+            }
+        }
+
+        // Configure metrics server
+        #[cfg(feature="metrics-server")] {
+            if let Some(metrics_config) = &config_file.metrics_server {
+                let bind_to = metrics_config.bind.as_ref()
+                    .and_then(|addr| addr.into_ip_address());
+
+                let credentials = metrics_config.password.as_ref().map(|password| {
+                    Credentials::new("metrics", password)
+                });
+
+                self.metrics_server = Some(Some(MetricsServerConfig {
+                    bind_to,
+                    port: metrics_config.port.unwrap_or(consts::METRICS_DEFAULT_PORT),
+                    credentials,
+                }));
+            }
+        }
+
+        // Configure custom seeds
+        for seed in &config_file.network.seed_nodes {
+            self.seed(Seed::try_from(seed.clone())
+                .map_err(|e| Error::config_error(format!("Invalid seed: {:?}: {}", seed, e)))?);
+        }
+
         // Configure validator
         if config_file.validator.is_some() {
```

Comment:
This probably needs to go behind a feature flag as the `validator` field itself is behind a feature flag, too.

Reply:
`ValidatorConfig {}` can be simplified to `ValidatorConfig`.